Remove usage of BuildConfig and ImageStream

lunika opened the pull request:
## Purpose

We don't want to use anymore the OC private registry. We want to remove all playbooks related to `BuildConfig` and `ImageStream` resources. Also remove this resources in `nextcloud` app


## Proposal

- [x] remove `BuildConfig` and `ImageStream` related playbooks
- [x] remove `BuildConfig` and `ImageStream` templates in `nextcloud` app

lunika committed:
🔥(nextcloud) remove usage of build images

Custom images for nextcloud are removed from openshift registry. By
default we use the fundocker/nextcloud image and for a custom one you
should craft your own and use nextcloud_image_name and
nextcloud_image_tag to select them.

lunika committed:
🔥(core) remove image related playbooks

We want to remove the usage of BuildConfig and ImageStreams. To do this
all playbooks related to these objects are deleted.

madmatah commented on tasks/get_objects_for_app.yml:
```diff
@@ -14,18 +14,15 @@
     endpoints: "{{ templates | map('regex_search', '.*/ep.*\\.yml\\.j2$') | select('string') | list }}"
     services: "{{ templates | map('regex_search', '.*/svc.*\\.yml\\.j2$') | select('string') | list }}"
     statefulsets: "{{ templates | map('regex_search', '.*/sts.*\\.yml\\.j2$') | select('string') | list }}"
-    streams: "{{ templates | map('regex_search', '.*/is.*\\.yml\\.j2$') | select('string') | list }}"
```
I love this PR ! But I see that the `forum` app is still declaring an image stream in `apps/forum/templates/services/app/is.yml.j2`.

But I can't figure out if it is still in use.

sampaccoud replied:
That's because I removed the BC and forgot about the IS :man_shrugging:

lunika replied:
I added a commit to remove this file in this PR

sampaccoud commented on apps/nextcloud/templates/services/nginx/dc.yml.j2:
```diff
@@ -69,7 +69,7 @@ spec:
             periodSeconds: 5
 
       initContainers:
-        - image: "{{ internal_docker_registry }}/{{ project_name }}/nextcloud:{{ nextcloud_image_tag }}"
```
missing imagePullSecrets?

lunika replied:
`imagePullSecrets` is at the same level, you define one for your deploymentconfig, is it possible to have several ?

sampaccoud replied:
The imagePullSecrets is a list so I suppose you can have several and they either match it by tld or by trying them successively? My point is that, in this case, the images used by the container and by the initContainer are different and come from different registries :man_shrugging:

lunika replied:
Can you give me the documentation ? In this schema I don't understand this is a list https://docs.openshift.com/container-platform/3.9/rest_api/oapi/v1.DeploymentConfig.html#object-schema
But I agree with you, here the image can come from different registries.

sampaccoud replied:
SO has this: https://stackoverflow.com/questions/54717135/are-multiple-imagepullsecrets-allowed-and-used-by-kubernetes-to-pull-an-image-fr It seems to be a feature for High Availability of docker registries but should also allow to pull images from different private registries like your present case.

lunika replied:
:pray: I made the modification

sampaccoud commented on bin/arnold:
```diff
@@ -598,33 +596,13 @@ function _vault(){
 }
 
-# create image streams for each application
-function create_image_streams() {
```
create_image_streams was duplicated!?

lunika replied:
yes

sampaccoud commented on tasks/get_objects_for_app.yml:
```diff
@@ -36,15 +33,6 @@
     - statefulset
     - stream
```
should you?

search for "stream" on the code base?

lunika replied:
Yes I should remove it, it's not used in the application.

lunika committed:
🔥(forum) remove ImageStream resource

An ImageStream resource was defined in the forum application but
unused. We must remove it.

madmatah reviewed (approved):
![](https://media.giphy.com/media/f5MyRwg36oClblXTVE/giphy.gif)

jmaupetit commented on CHANGELOG.md:
```diff
@@ -8,6 +8,11 @@ Versioning](http://semver.org/spec/v2.0.0.html).
 
+### Removed
+
+- Nextcloud : remove usage of BuildConfig and ImageStream
```
```suggestion
- Nextcloud: remove usage of BuildConfig and ImageStream objects
```

jmaupetit commented on CHANGELOG.md:
```diff
@@ -8,6 +8,11 @@ Versioning](http://semver.org/spec/v2.0.0.html).
+### Removed
+
+- Nextcloud : remove usage of BuildConfig and ImageStream
+- Build image and image stream playbooks.
```
```suggestion
- Build image and image stream playbooks
```

sampaccoud reviewed (approved):
<img src="https://media.giphy.com/media/mEzggrL7l4pbisoqTI/giphy.gif" width="300">
